DataStructures/HashTables/HashTable.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace DataStructures.HashTables
+{
+    public class HashTable<TValue> 
+    {
+        public HashTable(int buckets)
+        {
+            Buckets = buckets;
+            ResetBuckets();
+        }
+        public int Buckets { get; set; }
+        private static int BreakdownString(string key)
+        {
+            int x = 0;
+            for (int i = 0; i < key.Length; i++)
+            {
+                x = x + (int)key[i];
+            }
+            return x;
+        }
+        public static int GetHash(string key)
+        {
+            
+            int key_ascii = BreakdownString(key);
+            return key_ascii * 269;
+
+        }
+        public void Clear() => ResetBuckets();
+        private void ResetBuckets()
+        {
+            BucketList = new List<HashList<TValue>>();
+            for (int i = 1; i <= Buckets; i++)
+            {
+                HashList<TValue> bucket = new HashList<TValue>(i);
+                BucketList.Add(bucket);
+            }
+        }
+        public List<HashList<TValue>> BucketList { get; set; }
+        public TValue this[string key] { get => Get(key); }
+
+        public void Add(string key, TValue value)
+        {
+            int hash = GetHash(key);
+            int x = hash % Buckets;
+            var bucket = BucketList.Find(b=> b.Bucket == x);
```

Comment:
`Find()` is a O(n) scan; you want to access the bucket directly! Also, `Bucket` is 1-indexed, but `x` is 0-indexed.
```suggestion
            var bucket = BucketList[x];
```